rest_assured/testcases.py

```diff
@@ -50,28 +50,31 @@ def get_object(self, factory):
 
         return factory.create()
 
-    def setUp(self):
-        """Generates the main object and user instance if needed.
+    def get_user_factory(self):
+        """Generates the user instance if needed.
 
         The user instance will be created only if the ``user_factory`` attribute is set to the factory class.
 
         If there is an available user instance, that user will be force authenticated.
         """
-
         # create and force authenticate user
         user_factory = getattr(self, 'user_factory')
         if user_factory:
             self.user = user_factory.create()
             self.client.force_authenticate(self.user)
 
-        # create the object
-        self.object = self.get_object(self.get_factory_class())
+    def setUp(self):
+        """Generates the user instance if needed and the main object if required."""
+        self.get_user_factory()
```

Comment:
Just get back the `user_factory` and continue with generating user instance and authentication.